Core/Core/Grades/GradeListViewController.swift

```diff
@@ -134,6 +142,17 @@ public class GradeListViewController: ScreenViewTrackableViewController, Colored
         navigationController?.navigationBar.useContextColor(color)
     }
 
+    public override func viewDidAppear(_ animated: Bool) {
+        super.viewDidAppear(animated)
+
+        // Without this there was some weird empty space at the end of the tableview
+        // that went away after rotation or when we moved away from the screen and returned
+        if OfflineModeInteractorLive.shared.isOfflineModeEnabled() {
```

Comment:
Could we inject the instance when we create the view controller?

Reply:
Yes, I modified it.